rotkehlchen/chain/ethereum/modules/balancer/decoder.py

```diff
@@ -0,0 +1,300 @@
+                'amount_out': amount_out,
+            },
+        )
+
+        # When ETH is swapped it is wrapped to WETH and the ETH transfer happens before the SWAP
+        # event, we need to detect it we haven't done it yet.
+        if len(action_items) == 0 and from_token == self.weth.evm_address:
+            # when swapping eth the transfer event appears before the V2_SWAP event so we need
+            # to check if the asset swapped was ETH or not.
+            amount_of_eth = asset_normalized_value(
+                amount=amount_in,
+                asset=self.eth,
+            )
+            for event in decoded_events:
+                if (
+                    event.asset == A_ETH and event.balance.amount == amount_of_eth and
+                    event.event_type == HistoryEventType.SPEND and
+                    event.event_subtype == HistoryEventSubType.NONE
+                ):
+                    event.event_type = HistoryEventType.TRADE
+                    event.event_subtype = HistoryEventSubType.SPEND
+                    event.notes = f'Swap {event.balance.amount} {self.eth.symbol} in Balancer V2 from {event.location_label}'  # noqa: E501
+                    event.counterparty = CPT_BALANCER_V2
+
+        return None, [action_item]
+
+    def _maybe_enrich_balancer_v2_transfers(  # pylint: disable=no-self-use
+            self,
+            token: 'EvmToken',  # pylint: disable=unused-argument
+            tx_log: EvmTxReceiptLog,
+            transaction: EvmTransaction,
+            event: HistoryBaseEntry,
+            action_items: list[ActionItem],
+            all_logs: list[EvmTxReceiptLog],  # pylint: disable=unused-argument
+    ) -> bool:
+        """
+        Enrich tranfer transactions to address for swaps in balancer v2 protocol.
+        May raise:
+        - UnknownAsset
+        - WrongAssetType
+        """
+        if action_items is None or len(action_items) == 0 or transaction.to_address != VAULT_ADDRESS:  # noqa: E501
+            return False
+
+        if (
+            action_items[-1].extra_data is None or
+            'amount_out' not in action_items[-1].extra_data or
+            'to_token' not in action_items[-1].extra_data
+        ):
+            return False
+
+        asset = event.asset.resolve_to_evm_token()
+        to_amount = asset_normalized_value(
+            amount=action_items[-1].extra_data['amount_out'],
+            asset=asset,
+        )
+
+        if (
+            action_items[-1].extra_data['to_token'] != tx_log.address or
+            event.balance.amount != to_amount
+        ):
+            return False
+
+        event.counterparty = CPT_BALANCER_V2
+        event.event_type = HistoryEventType.TRADE
+        if asset.evm_address == action_items[-1].extra_data['to_token']:
+            event.event_subtype = HistoryEventSubType.RECEIVE
+            event.notes = f'Receive {event.balance.amount} {asset.symbol} in Balancer V2 from {event.location_label}'  # noqa: E501
```

Comment:
How does this read. I don't get it. `Receive 10 GNO in Balancer V2 from 0xmyaddress`?

This sounds wrong.

Shouldnt' it be `Receive 10 GNO from Balancer v2` or ... if anything more have a pool or something?

Also I see you use capitalized V2. Don't do it. Change it everywhere. It should be v2.

Reply:
Yes, as discussed changed it to `Receive 10 GNO from Balancer v2`